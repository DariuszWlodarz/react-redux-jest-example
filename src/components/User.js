import React from 'react';
import PropTypes from 'prop-types';


const User = ({user, onClickHandler}) => (
  <div onClick={() => onClickHandler(user)}>{`${user.id} ${user.email}`}</div>
);

User.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired
  }).isRequired,
  onClickHandler: PropTypes.func.isRequired
};

export default User;