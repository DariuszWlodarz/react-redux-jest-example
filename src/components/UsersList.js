import React from 'react';
import PropTypes from 'prop-types';
import User from './User';

const UsersList = ({users, onUserClickHandler}) => (
  <div>
    {users.map(user => (
      <User user={user} onClickHandler={onUserClickHandler} key={user.id} />
    ))}
  </div>
);

UsersList.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired
  })).isRequired,
  onUserClickHandler: PropTypes.func.isRequired
};

export default UsersList;