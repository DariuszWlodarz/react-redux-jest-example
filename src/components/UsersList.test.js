import React from 'react';
import UsersList from './UsersList';
import User from './User';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const users = [
  {
    id: 1,
    email: 'user1@example.com'
  },
  {
    id: 2,
    email: 'user2@example.com'
  }
];

test('snapshot', () => {
  const component = shallow(
    <UsersList users={users} onUserClickHandler={() => {}} />
  );

  expect(toJson(component)).toMatchSnapshot();
});

test('length', () => {
  let component = shallow(<UsersList users={users} onUserClickHandler={() => {}} />);

  expect(component.find(User)).toHaveLength(2);

  component = shallow(<UsersList users={[]} onUserClickHandler={() => {}} />);

  expect(component.find(User)).toHaveLength(0);
});

