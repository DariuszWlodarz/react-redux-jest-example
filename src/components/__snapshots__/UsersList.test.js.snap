// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <User
    onClickHandler={[Function]}
    user={
      Object {
        "email": "user1@example.com",
        "id": 1,
      }
    }
  />
  <User
    onClickHandler={[Function]}
    user={
      Object {
        "email": "user2@example.com",
        "id": 2,
      }
    }
  />
</div>
`;
